### PR TITLE
docs: remove redundant field in Device CRD

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### Ⅰ. Describe what this PR does
There are three changes in this PR:

1. 
```
type DevicesSpec struct {
    Items  []Device `json:"items"`
}
```
was removed

2.
`type DevicesStatus struct {}` was renamed to `type DeviceStatus struct {}`

3.
`ID` in `type DeviceInfo struct {}` was changed to `UUID`.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md [hunk 1]
```diff
@@ -303,61 +303,57 @@ Implements a new component called  `Device Reporter` in koordlet to create or up
...
 }
 
-type Device struct {
-    metav1.TypeMeta   `json:",inline"`
-    metav1.ObjectMeta `json:"metadata,omitempty"`
-
-    Spec   DeviceSpec `json:"spec,omitempty"`
-    Status DevicesStatus `json:"status,omitempty"`
+type DeviceStatus struct {
+	Allocations []DeviceAllocation `json:"allocations"`
```
Comment (hormes) [inline docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:329]: Why register the allocation status here?

Comment (jasonliu747) [reply docs/proposals/scheduling/20220629-fine-grained-device-scheduling.md:329]: I didn't bring Allocations into DeviceStatus in this PR. For more detail, you might wanna check with @buptcozy.

**Pull request merged**